src/lib.rs

```diff
@@ -1453,4 +1453,52 @@ mod tests {
             .map(|i| u8::from_str_radix(&s[i..i + 2], 16).unwrap())
             .collect()
     }
+
+    #[test]
+    fn check_payload_length_exactly_32_bytes() {
```

Comment:
Just to make sure, you omitted leaf with length < 32 bytes on purpose?

Reply:
Nope,  but I had a merge mishap so I assume it got lost during that time. I have also lost a fix for the empty leaf issue :sweat: 

Anyhow, I'll rewrite it as it shouldn't be too long.